ouroboros-network/src/Ouroboros/Network/Protocol/ChainSync/ClientPipelined.hs

```diff
@@ -179,63 +252,68 @@ data ChainSyncInstruction header point tip
 chainSyncClientPeerPipelined
     :: forall header point tip m a.
        Monad m
-    => ChainSyncClientPipelined header point tip m a
+    => Tracer m TraceChainsyncClientObservation
+    -> ChainSyncClientPipelined header point tip m a
     -> PeerPipelined (ChainSync header point tip) AsClient StIdle m a
 
-chainSyncClientPeerPipelined (ChainSyncClientPipelined mclient) =
-    PeerPipelined $ SenderEffect $ chainSyncClientPeerSender Zero <$> mclient
+chainSyncClientPeerPipelined tracer (ChainSyncClientPipelined mclient) =
+    PeerPipelined $ SenderEffect $ chainSyncClientPeerSender tracer Zero <$> mclient
 
 
 chainSyncClientPeerSender
     :: forall n header point tip m a.
        Monad m
-    => Nat n
+    => Tracer m TraceChainsyncClientObservation
+    -> Nat n
     -> ClientPipelinedStIdle n header point tip m a
     -> PeerSender (ChainSync header point tip)
                   AsClient StIdle n
                   (ChainSyncInstruction header point tip)
                   m a
 
-chainSyncClientPeerSender n@Zero (SendMsgRequestNext stNext stAwait) =
-
-    SenderYield
+chainSyncClientPeerSender tracer n@Zero (SendMsgRequestNext stNext stAwait) = SenderEffect $ do
```

Comment:
It will clutter the diff (so do that in a separate patch): it will look better if we re-indent the code and put `SenderEffect` in a new line.

Reply:
Will place each argument on its own line instead in order to avoid indenting the function one step further.